src/main/java/org/terasology/climateConditions/alterationEffects/BodyTemperatureAlterationSystem.java

```diff
@@ -0,0 +1,123 @@
+
+// Copyright 2020 The Terasology Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+package org.terasology.climateConditions.alterationEffects;
+
+import org.terasology.alterationEffects.AlterationEffect;
+import org.terasology.alterationEffects.AlterationEffects;
+import org.terasology.alterationEffects.OnEffectRemoveEvent;
+import org.terasology.climateConditions.AffectBodyTemperatureEvent;
+import org.terasology.context.Context;
+import org.terasology.entitySystem.Component;
+import org.terasology.entitySystem.entity.EntityRef;
+import org.terasology.entitySystem.event.ReceiveEvent;
+import org.terasology.entitySystem.systems.BaseComponentSystem;
+import org.terasology.entitySystem.systems.RegisterMode;
+import org.terasology.entitySystem.systems.RegisterSystem;
+import org.terasology.logic.delay.DelayedActionTriggeredEvent;
+import org.terasology.registry.In;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.regex.Pattern;
+
+/**
+ * This system manages the effects of the Body Temperature Alteration Effect.
+ */
+@RegisterSystem(value = RegisterMode.AUTHORITY)
+public class BodyTemperatureAlterationSystem extends BaseComponentSystem {
+    public static final String BODY_TEMPERATURE = "BodyTemperature";
+
+    @In
+    Context context;
+
+    /**
+     * This will store the mapping of the effect constants to the effect components.
+     */
+    private Map<String, Class<? extends Component>> effectComponents = new HashMap<>();
+
+    /**
+     * This will store the mapping of the effect constants to the alteration effects.
+     */
+    private Map<String, AlterationEffect> alterationEffects = new HashMap<>();
+
+    @Override
+    public void initialise() {
+        effectComponents.put(BODY_TEMPERATURE, AffectBodyTemperatureComponent.class);
+        alterationEffects.put(BODY_TEMPERATURE, new BodyTemperatureAlterationEffect(context));
+    }
+
+    /**
+     * When the body temperature of an entity changes, modify the change in temperature based on what effects are being
+     * applied to the entity.
+     *
+     * @param event Stores information on the body temperature change that is to be applied and collects
+     *         potential modifiers.
+     * @param entityRef The entity experiencing a change in body temperature.
+     */
+    @ReceiveEvent
+    public void modifyTemperatureChange(AffectBodyTemperatureEvent event, EntityRef entityRef) {
+        // If the entity's stunned, prevent it from moving.
+        AffectBodyTemperatureComponent affectBodyTemperatureComponent =
+                entityRef.getComponent(AffectBodyTemperatureComponent.class);
+        if (affectBodyTemperatureComponent != null) {
+            if (affectBodyTemperatureComponent.id.equals("")) {
+                event.addPostMultiply(affectBodyTemperatureComponent.postMultiplier);
+            } else {
+                if (affectBodyTemperatureComponent.id.equals("activateWhenEnvironmentColder") && event.isNegative()) {
```

Comment:
Here it would also be better to not type out the string again, but use a common constant. Using an enum would also help, I think.